### ISSUE TITLE
Compliance checker errors on valid CF-1.6 trajectory file

### ISSUE THREAD

**Issue by rsignell-usgs** [ioos/compliance-checker#281]
I fixed up an OOI Glider netcdf file to be CF-1.6 `featureType=trajectory` and verified that it loads correctly into ToolsUI (https://www.youtube.com/watch?v=43eCCVbpGbg).  The file is here:
http://geoport-dev.whoi.edu/thredds/catalog/usgs/data2/rsignell/test/catalog.html?dataset=usgs/data2/rsignell/test/ooi_glider.nc

and if we load the OPeNDAP URL
http://geoport-dev.whoi.edu/thredds/dodsC/usgs/data2/rsignell/test/ooi_glider.nc.html
into the compliance checker and check for CF-1.6 compliance, we get lots of messages like:

```
Name    Priority    Corrective action
2   Coordinate time's dimension, obs, is not a coordinate variable
```

report is here: 
http://data.ioos.us/compliance/report/149219b60500f9ed516deefb6ab3d965d68eb5b3

**Comment by lukecampbell** [comment]
Yeah, this has to do with how the current version of the CF Checker detects features, which is wrong. I need to do a major refactor of the CF checker and integrate all of the feature detection logic I implemented for the NCEI checkers back into the main CF Checker. 

It does ok for time-series and stations, but for more complicated feature types, I've observed very questionable behavior. It seems to think that a lot of these like ragged arrays are grids and so it tries to treat it like a "reduced horizontal grid", which it almost never is.

**Comment by rsignell-usgs** [comment]
@lukecampbell , yes, this is a straight-up trajectory:  just a bunch of obs along a x,y,z,t path.

**Comment by lukecampbell** [comment]
For the time being, I recommend to people to use the NCEI checkers as they provide better feature detection and better written guidance for corrective actions. But this issue you describe is my number one priority for 2.3

**Comment by benjwadams** [comment]
Unable to reproduce against current master.  I think this was fixed by one of the PRs but can't remember which one, but I'm not going to go back and run `git bisect` to find out.

**Comment by benjwadams** [comment]
Is this now fixed from #428?
